riot_api/methods/get_rank.py

```diff
@@ -0,0 +1,132 @@
+"""
+Data processing the data from riot API
+"""
+import pydash
+from db.db import Session
+from db.models.summoners import Summoners
+
+from utils.utils import get_file_path
+
+from .. import watcher, MY_REGION
+
+
+session = Session()
+
+# Get summoner rank.
+def get_summoner_rank(name: str):
+    """Gets the summoner's rank information from riot watcher api
+    Parameters:
+    name (str): name of the summoner
+
+    Returns:
+    summoner_profile (dict): rank information about the summoner
+
+    """
+    # TODO BEFORE REQUESTING DATA, CHECK DB AND UPDATE_TIME TO SEE IF WE ALREADY HAVE DATA.
+    # IF WE HAVE DATA BUT UPDATE_TIME IS OVER THE CONSTRAINT, WE SHOULD DELETE THE ROW.
+
+    # We need to get id
+    user = watcher.summoner.by_name(MY_REGION, name)
+
+    # First check if we have existing record for given summoner name
+    # Create query; TODO: check for update_time
+    summoner_cached_query = session.query(Summoners).filter(
+        Summoners.summoner_name == user["name"]
+    )
+
+    # Execute query
+    summoner_cached = summoner_cached_query.one_or_none()
+
+    # If data exists, form data and return here.
+    if summoner_cached:
+        summoner_cached = dict(summoner_cached.__dict__)
+        tier = " ".join(
+            [summoner_cached["tier_division"], summoner_cached["tier_rank"]]
+        )
+
+        emblem_path = get_file_path(
+            f"images/Emblem_{summoner_cached['tier_division'].capitalize()}.png"
+        )
+
+        summoner_profile = {
+            "user_name": summoner_cached["summoner_name"],
+            "summoner_icon_image_url": summoner_cached["summoner_icon_image_url"],
+            "summoner_level": summoner_cached["summoner_level"],
+            "tier_image_path": emblem_path,
+            "tier_image_name": f"Emblem_{summoner_cached['tier_division'].capitalize()}.png",
+            "tier": tier,
+            "puuid": summoner_cached["puuid"],
+            "tier_division": summoner_cached["tier_division"],
+            "tier_rank": summoner_cached["tier_rank"],
+            "solo_win": summoner_cached["solo_win"],
+            "solo_loss": summoner_cached["solo_loss"],
+            "league_points": summoner_cached["league_points"],
+        }
+
+        return summoner_profile
+    # Cached value doesn't exist; Grab data from API.
```

Comment:
What's the reason to not use `else` here?
Ex: `if summoner_cached is None:`

Reply:
What's the need of else? 
we are returning on line 66 if cached value exists